src/middlewared/middlewared/plugins/network.py

```diff
@@ -2082,6 +2085,19 @@ def ip_in_use(self, choices=None):
         if not choices['loopback']:
             ignore_nics.append('lo')
         ignore_nics = tuple(ignore_nics)
+        static_ips = {}
+        if choices['static']:
+            for i in self.middleware.call_sync('interface.query'):
+                for alias in i.get('failover_virtual_aliases') or []:
+                    static_ips[alias['address']] = alias['address']
+                for alias in (
+                    i['aliases'] if (
+                        not self.middleware.call_sync('failover.config')['disabled'] and self.middleware.call_sync(
```

Comment:
Two things:

1. calling call_sync for every alias for every interface
2. checking only once for `failover.licensed` is good enough